In [1]:
# Import libraries
import pandas as pd
import numpy as np
import os

### 1. Import the data sets into Jupyter

In [2]:
# Tell Python to remember a main folder path
path = r'C:\Users\Shruti Kolhatkar\Documents\Instacart Basket Analysis_ShrutiK'

In [3]:
df_prods = pd.read_csv(os.path.join(path, '02 Data', 'Prepared Data', 'products_checked.csv'), index_col = False)


In [4]:
df_ords_combined=  pd.read_pickle(os.path.join(path, '02 Data', 'Prepared Data', 'orders_products_combined.pkl'))


### 2. Check the dimension of the imported dataframes

In [5]:
# Check the output
df_prods.head()


Unnamed: 0  product_id                                       product_name  \
0           0           1                         Chocolate Sandwich Cookies   
1           1           2                                   All-Seasons Salt   
2           2           3               Robust Golden Unsweetened Oolong Tea   
3           3           4  Smart Ones Classic Favorites Mini Rigatoni Wit...   
4           4           5                          Green Chile Anytime Sauce   

   aisle_id  department_id  prices  
0        61             19     5.8  
1       104             13     9.3  
2        94              7     4.5  
3        38              1    10.5  
4         5             13     4.3

In [7]:
df_prods.describe()


Unnamed: 0    product_id      aisle_id  department_id        prices
count  49672.000000  49672.000000  49672.000000   49672.000000  49672.000000
mean   24852.005053  24850.349775     67.762442      11.728942      9.993282
std    14342.265579  14340.705287     38.315784       5.850779    453.615536
min        0.000000      1.000000      1.000000       1.000000      1.000000
25%    12432.750000  12432.750000     35.000000       7.000000      4.100000
50%    24851.500000  24850.500000     69.000000      13.000000      7.100000
75%    37272.250000  37268.250000    100.000000      17.000000     11.100000
max    49692.000000  49688.000000    134.000000      21.000000  99999.000000

In [8]:
df_prods.shape


(49672, 6)

In [13]:
df_prods


product_id                                       product_name  \
0               1                         Chocolate Sandwich Cookies   
1               2                                   All-Seasons Salt   
2               3               Robust Golden Unsweetened Oolong Tea   
3               4  Smart Ones Classic Favorites Mini Rigatoni Wit...   
4               5                          Green Chile Anytime Sauce   
...           ...                                                ...   
49667       49684          Vodka, Triple Distilled, Twist of Vanilla   
49668       49685                 En Croute Roast Hazelnut Cranberry   
49669       49686                                   Artisan Baguette   
49670       49687         Smartblend Healthy Metabolism Dry Cat Food   
49671       49688                             Fresh Foaming Cleanser   

       aisle_id  department_id  prices  
0            61             19     5.8  
1           104             13     9.3  
2            94              7     4.5  
3            38              1    10.5  
4             5             13     4.3  
...         ...            ...     ...  
49667       124              5     5.3  
49668        42              1     3.1  
49669       112              3     7.8  
49670        41              8     4.7  
49671        73             11    13.5  

[49672 rows x 5 columns]

In [14]:
df_ords_combined.head()


order_id  user_id  order_number  orders_day_of_week  order_hour_of_day  \
0   2539329        1             1                   2                  8   
1   2539329        1             1                   2                  8   
2   2539329        1             1                   2                  8   
3   2539329        1             1                   2                  8   
4   2539329        1             1                   2                  8   

   days_since_prior_order  first_order  product_id  add_to_cart_order  \
0                     NaN         True         196                  1   
1                     NaN         True       14084                  2   
2                     NaN         True       12427                  3   
3                     NaN         True       26088                  4   
4                     NaN         True       26405                  5   

   reordered _merge  
0          0   both  
1          0   both  
2          0   both  
3          0   both  
4          0   both

In [16]:
df_ords_combined = df_ords_combined.drop(['_merge'], axis=1)

In [19]:
df_ords_combined.shape

(32434489, 10)

In [17]:
df_ords_prods_merge = df_prods.merge(df_ords_combined, on = 'product_id', indicator = True)


In [20]:
df_ords_prods_merge.head()


product_id                product_name  aisle_id  department_id  prices  \
0           1  Chocolate Sandwich Cookies        61             19     5.8   
1           1  Chocolate Sandwich Cookies        61             19     5.8   
2           1  Chocolate Sandwich Cookies        61             19     5.8   
3           1  Chocolate Sandwich Cookies        61             19     5.8   
4           1  Chocolate Sandwich Cookies        61             19     5.8   

   order_id  user_id  order_number  orders_day_of_week  order_hour_of_day  \
0   3139998      138            28                   6                 11   
1   1977647      138            30                   6                 17   
2    389851      709             2                   0                 21   
3    652770      764             1                   3                 13   
4   1813452      764             3                   4                 17   

   days_since_prior_order  first_order  add_to_cart_order  reordered _merge  
0                     3.0        False                  5          0   both  
1                    20.0        False                  1          1   both  
2                     6.0        False                 20          0   both  
3                     NaN         True                 10          0   both  
4                     9.0        False                 11          1   both

In [21]:
df_ords_prods_merge.shape

(32404859, 15)

In [22]:
df_ords_prods_merge['_merge'].value_counts()

both          32404859
right_only           0
left_only            0
Name: _merge, dtype: int64

### 4. Use the indicator argument to check whether there was a full match between the two dataframes

In [15]:
df_merged_large['_merge'].value_counts()

both          32434489
right_only           0
left_only            0
Name: _merge, dtype: int64

#### The resulting dataframe (after the merge) has 32,434,489 rows, and each of those rows have information found in both input data sets, as we used an inner join for the purposes of this project.

### 5. Export data in pickle format

In [23]:
df_ords_prods_merge.to_pickle(os.path.join(path, '02 Data','Prepared Data', 'ords_prods_merge.pkl'))